lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -8727,6 +8735,14 @@ namespace Js
 #endif
                 }
             }
+            else
+            {
+#ifdef DEBUG
+                for (int __i = i; __i < EQUIVALENT_TYPE_CACHE_SIZE; __i++)
```

Comment:
I dont know this code very well to ascertain that this assert is valid. Maybe @pleath has more context.

Reply:
The loop *should* be the least-often-executed part of the code. If we're spending a lot of time there, then we're seeing many new (probably non-shared) types, and/or we keep seeing types that were evicted from the cache. Do you know which it is?

Reply:
stumbled into this with acme-air run. Couple of loops here and there. Not all of them are hot.`JavascriptNativeOperators::CheckIfTypeIsEquivalent` is more common (hence the changes on this PR)

Reply:
I see. So you're saying that the routine is hot, but not necessarily the loop? The reason I ask is that it would be interesting to confirm that we keep taking the slow path through the routine and, if so, to understand why, since it may indicate that we're failing to share types.

Reply:
Yes, `JavascriptNativeOperators::CheckIfTypeIsEquivalent` is hot and I made the most changes there. I will take a look but for the time being, could you please review the PR or do you think this PR is good to go?

Reply:
I think it's fine, but please let me know in an e-mail how to run the particular scenario in which the method shows up as hot so I can investigate for type-sharing purposes. (Thanks.)